python/data_structures/linked_list.py

```diff
@@ -11,17 +15,27 @@ def insert(self, value):
 
 
     def __str__(self):
-        str = "NULL"
         current = self.head
+        str = ""
         while current:
-            if current.next:
-                str = f"{{ {current.value} }} -> {{ {current.next.value} }} -> " + str
-            else:
-                if str == "NULL":
-                    str = f"{{ {current.value} }} -> " + str
+            str += f"{{ {current.value} }} -> "
             current = current.next
+        str += "NULL"
         return str
```

Comment:
Hooray for refactoring! 🚀 Don't be afraid to include this in its own commit. It may not be part of the task at hand, but adding a commit for "refactors LinkedList.__str__ for conciseness" is better than having the change sneak in somewhere else.